File: java/src/main/java/cn/az/code/may/RansomNote.java
package cn.az.code.may;

/**
 * @author az
 * @date 5/3/2020
 */
public class RansomNote {

    public static void main(String[] args) {
        RansomNote rn = new RansomNote();
        System.out.println(rn.canConstruct("aa", "aab"));
    }

    public boolean canConstruct(String ransomNote, String magazine) {
        char[] notes = ransomNote.toCharArray(), magazines = magazine.toCharArray();
        int find = 0;
        for (char c : notes) {
            int count = 0;
            for (int i = 0; i < magazines.length; i++) {
                if (magazines[i] == c) {
                    magazines[i] = '0';
                    find++;
                    break;
                } else {
                    if (count == magazines.length) {
                        return false;
                    }
                    count++;
                }
            }
        }
        return find == notes.length;
    }
}

File: archive/go/2020/may/RansomNote.go
package may

func CanConstruct(ransomNote string, magazine string) bool {
	find := 0
	for _, c1 := range ransomNote {
		count := 0
		for _, c2 := range magazine {
			if c2 == c1 {
				c2 = '0'
				find++
				break
			} else {
				if count == len(magazine) {
					return false
				}
				count++
			}
		}
	}
	return find == len(ransomNote)
}
